File: DiskTudoCliente.APP/src/app/_models/Pedido.ts
import { Produto } from './Produto';
import { CarrinhoItemModel } from './CarrinhoItemModel';

export class Pedido {
    id: number;
    dataHora: Date;
    valorTotal: number;
    userId?: number;
    item: CarrinhoItemModel[];
    endereco: string;
    telefone: string;

    constructor() {
        this.item = [];
    }
}

File: DiskTudo.Domain/Pedido.cs
using System;
using System.Collections.Generic;
using DiskTudo.Domain.Identity;

namespace DiskTudo.Domain
{
    public class Pedido
    {
        public int Id { get; set; }
        public DateTime DataHora { get; set; }
        public int UserId { get; set; }

        public decimal ValorTotal { get; set; }

        public string Item { get; set; }
        public string Endereco { get; set; }
        public string Telefone { get; set; }
    }
}